lib/helpers/helper.routes.ts

```diff
@@ -2,18 +2,17 @@ type RouteParams = { [key: string]: string | number }
 type QueryParams = { [key: string]: string | number }
 
 export function buildRoute({
-  route,
   params,
   query,
 }: {
-  route: string
   params?: RouteParams
   query?: QueryParams
 }): string {
+  let route = ""
   if (params) {
     route = Object.keys(params).reduce((acc, key) => {
       const value = encodeURIComponent(params[key])
-      return acc.replace(`:${key}`, value)
+      return `${acc}/${value}`
```

Comment:
There is something about the string concatenation handling here.
You should be able to solve the url creation alone by using [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) and [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL) and eg: `URL.searchParams.append`
Maybe I am wrong.